data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/ProcessWorker.java

```diff
@@ -58,7 +56,7 @@ public void run() {
                     postToSink(records); //TODO use the response to ack the buffer on failure?
                 }
                 // Checkpoint the current batch read from the buffer after being processed by prepper and sinks.
-                readBuffer.checkpoint(checkpointState);
+                readBuffer.checkpoint();
```

Comment:
this works because we currently do not worry about partially processed records - per design we expect some preppers to drop some records [based on context - it may mean they are processed] and we indefinitely wait for sinks to post and fail even if one of the sink fails. I do not see we changing any of this, so we are good here